### DISCUSSION TITLE
Adding provenance metadata to external datasets

### DISCUSSION THREAD

**Discussion by NicolasRouquette** [NASA-AMMOS/aerie#1211] [Ideas]
Aerie has the capability to receive timeline data from an external source via the concept of 'external dataset': https://nasa-ammos.github.io/aerie-docs/planning/external-datasets/

Unfortunately, this capability lacks provenance metadata information about the external source. This is unfortunate because it prevents some interesting use cases like the one described below.

Aerie produces timelines that we export to a CSV file to drive two external simulators, Sim1 and Sim2, whose outputs we import into Aerie as external datasets. It would be helpful to show in the Aerie UI the provenance of each timeline (Aerie, Sim1, Sim2). It would be helpful to have the flexibility of toggling the display of datasets (e.g., only Aerie, Aerie + Sim1, Aerie + Sim2, All).

**Comment by mattdailis** [comment]
Hi Nicolas, I think this is an excellent suggestion. It is possible to encode the provenance of a resource using a naming convention (e.g. `/sim1/subsystem/resource-name` and `/sim2/subsystem/resource-name`), but Aerie should additionally provide the id of the source dataset. If Aerie allowed giving the datasets themselves meaningful names, that would also help make it easier to manage them
